pkg/utils/common_test.go

```diff
@@ -0,0 +1,622 @@
+						Version:   "v1",
+						Kind:      "StatefulSet",
+						Name:      "statefulset1",
+						Namespace: "default",
+						Envelope:  nil,
+					},
+					Condition: metav1.Condition{
+						Type:               fleetv1beta1.WorkConditionTypeAvailable,
+						Status:             metav1.ConditionFalse,
+						ObservedGeneration: 0,
+						LastTransitionTime: time1,
+						Reason:             "WorkNotAvailableYet",
+						Message:            "message1",
+					},
+				},
+			},
+			new: []fleetv1beta1.FailedResourcePlacement{
+				{
+					ResourceIdentifier: fleetv1beta1.ResourceIdentifier{
+						Group:     "apps",
+						Version:   "v1",
+						Kind:      "Deployment",
+						Name:      "deployment1",
+						Namespace: "default",
+						Envelope: &fleetv1beta1.EnvelopeIdentifier{
+							Name:      "test-envelope-object",
+							Namespace: "default",
+							Type:      fleetv1beta1.ConfigMapEnvelopeType,
+						},
+					},
+					Condition: metav1.Condition{
+						Type:               fleetv1beta1.WorkConditionTypeApplied,
+						Status:             metav1.ConditionFalse,
+						ObservedGeneration: 1,
+						LastTransitionTime: time2,
+						Reason:             "ManifestApplyFailed",
+						Message:            "message2",
+					},
+				},
+			},
+			want: false,
+		},
+		{
+			name: "compare two non-equal failed resource placements of different length, new is empty",
+			old: []fleetv1beta1.FailedResourcePlacement{
+				{
+					ResourceIdentifier: fleetv1beta1.ResourceIdentifier{
+						Group:     "apps",
+						Version:   "v1",
+						Kind:      "Deployment",
+						Name:      "deployment1",
+						Namespace: "default",
+						Envelope: &fleetv1beta1.EnvelopeIdentifier{
+							Name:      "test-envelope-object",
+							Namespace: "default",
+							Type:      fleetv1beta1.ConfigMapEnvelopeType,
+						},
+					},
+					Condition: metav1.Condition{
+						Type:               fleetv1beta1.WorkConditionTypeApplied,
+						Status:             metav1.ConditionFalse,
+						ObservedGeneration: 0,
+						LastTransitionTime: time1,
+						Reason:             "ManifestApplyFailed",
+						Message:            "message1",
+					},
+				},
+				{
+					ResourceIdentifier: fleetv1beta1.ResourceIdentifier{
+						Group:     "apps",
+						Version:   "v1",
+						Kind:      "StatefulSet",
+						Name:      "statefulset1",
+						Namespace: "default",
+						Envelope:  nil,
+					},
+					Condition: metav1.Condition{
+						Type:               fleetv1beta1.WorkConditionTypeAvailable,
+						Status:             metav1.ConditionFalse,
+						ObservedGeneration: 0,
+						LastTransitionTime: time1,
+						Reason:             "WorkNotAvailableYet",
+						Message:            "message1",
+					},
+				},
+			},
+			new:  []fleetv1beta1.FailedResourcePlacement{},
+			want: false,
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			got := IsFailedResourcePlacementsEqual(tc.old, tc.new)
+			if diff := cmp.Diff(tc.want, got); diff != "" {
```

Comment:
Also a nit, but since the diff is just comparing two bool values, it might be better to just use a equality check here